### PR TITLE
GPC-153 Move to v0.3 of linter and fix errors and warnings

### PR DESCRIPTION
**Pull request by Mjwillis**

### CODE REVIEW

File: src/main/kotlin/uk/gov/gdx/datashare/config/OpenApiConfiguration.kt [hunk 1]
```diff
@@ -30,11 +32,27 @@ class OpenApiConfiguration(
...
     .info(
       Info().title("GDX Data Share API")
         .version(version)
+        .license(License().name("MIT").url("https://opensource.org/license/mit-0"))
         .description("API for obtaining data about citizen life events")
-        .contact(Contact().name("GDX Vision Team").email("mike.willis@digital.cabinet-office.gov.uk")),
+        .contact(
+          Contact()
+            .name("GDX Platform Team")
+            .email("gdx-dev-team@digital.cabinet-office.gov.uk")
```
Comment (ethanmills) [inline src/main/kotlin/uk/gov/gdx/datashare/config/OpenApiConfiguration.kt:53]: Does this address work externally yet?

File: src/main/kotlin/uk/gov/gdx/datashare/config/OpenApiConfiguration.kt [hunk 2]
```diff
@@ -30,11 +32,27 @@ class OpenApiConfiguration(
...
       ),
     )
+    .tags(
+      listOf(
+        Tag().name("01. Acquirer").description("Acquiring events from the platform"),
+        Tag().name("02. Supplier").description("Supplying events to the platform"),
+        Tag().name("11. Acquirers").description("Maintenance of Acquirers"),
+        Tag().name("12. Suppliers").description("Maintenance of Suppliers"),
+        Tag().name("13. Admin").description("Admin functions"),
+        Tag().name("20. Data").description("Maintenance of Data"),
```
Comment (ethanmills) [inline src/main/kotlin/uk/gov/gdx/datashare/config/OpenApiConfiguration.kt:42]: I wonder if we can do some clever reflection so we don't need to duplicate this list

**Review by ethanmills — APPROVED** [review]
These new linter rules seem quite restrictive and angry

File: src/main/kotlin/uk/gov/gdx/datashare/config/OpenApiConfiguration.kt (hunk 1 above)
Comment (oskarwilliams378) [reply src/main/kotlin/uk/gov/gdx/datashare/config/OpenApiConfiguration.kt:53]: I don't think so, I'm gonna poke Farrah again soon

File: src/main/kotlin/uk/gov/gdx/datashare/config/OpenApiConfiguration.kt [hunk 3]
```diff
@@ -30,11 +32,27 @@ class OpenApiConfiguration(
...
+      Info().title("GDX Platform API")
         .version(version)
+        .license(License().name("MIT").url("https://opensource.org/license/mit-0"))
         .description("API for obtaining data about citizen life events")
-        .contact(Contact().name("GDX Vision Team").email("mike.willis@digital.cabinet-office.gov.uk")),
+        .contact(
+          Contact()
+            .name("GDX Platform Team")
+            .email("gdx-dev-team@digital.cabinet-office.gov.uk")
+            .url("https://www.gov.uk/government/organisations/government-digital-service"),
```
Comment (Mjwillis) [inline src/main/kotlin/uk/gov/gdx/datashare/config/OpenApiConfiguration.kt:54]: Couldn't think of a better URL for this yet

**Pull request merged**